spec/support/logstash_test.rb

```diff
@@ -21,3 +21,13 @@ def random_port
     end
   end
 end
+
+def testing_thread(&block)
+  th = Thread.new do
+    block.call
+  end
+  #
+  # wait for the thread to spin
+  sleep(0.1)
```

Comment:
its only a test helper to make sure the thread is ready before doing any tests on it.